egs/wham/wavesplit/train.py

```diff
@@ -0,0 +1,218 @@
+import os
+import argparse
+import json
+
+import torch
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.data import DataLoader
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping
+from model import Wavesplit
+from dataloading import WHAMID
+
+from asteroid.engine.optimizers import make_optimizer
+from asteroid.engine.system import System
+from losses import ClippedSDR, SpeakerVectorLoss
+
+# Keys which are not in the conf.yml file can be added here.
+# In the hierarchical dictionary created when parsing, the key `key` can be
+# found at dic['main_args'][key]
+
+# By default train.py will use all available GPUs. The `id` option in run.sh
+# will limit the number of available GPUs for train.py .
+parser = argparse.ArgumentParser()
+parser.add_argument("--exp_dir", default="exp/tmp", help="Full path to save best validation model")
+
+
+class WavesplitTrainer(System):
+    def on_train_start(self) -> None:
+        self.loss_func["spk_loss"] = self.loss_func["spk_loss"].to(self.device)
+
+    def on_validation_epoch_start(self) -> None:
+        self.loss_func["spk_loss"] = self.loss_func["spk_loss"].to(self.device)
+
+    def training_step(self, batch, batch_nb):
+        mixtures, oracle_s, oracle_ids = batch
+        b, n_spk, frames = oracle_s.size()
+
+        # spk_vectors = self.model.get_speaker_vectors(mixtures)
```

Comment:
Here and in validation steps I use oracle embeddings for now and no speaker stack